python-frontend/src/main/java/org/sonar/plugins/python/api/PythonVisitorContext.java

```diff
@@ -23,53 +23,45 @@
 import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
-import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 import org.sonar.plugins.python.api.PythonCheck.PreciseIssue;
+import org.sonar.plugins.python.api.caching.CacheContext;
 import org.sonar.plugins.python.api.tree.FileInput;
 import org.sonar.python.semantic.ProjectLevelSymbolTable;
 import org.sonar.python.semantic.SymbolTableBuilder;
 
-public class PythonVisitorContext {
+public class PythonVisitorContext extends PythonInputFileContext {
 
   private final FileInput rootTree;
-  private final PythonFile pythonFile;
-  private File workingDirectory = null;
   private final RecognitionException parsingException;
   private List<PreciseIssue> issues = new ArrayList<>();
 
-
   public PythonVisitorContext(FileInput rootTree, PythonFile pythonFile, @Nullable File workingDirectory, @Nullable String packageName) {
+    super(pythonFile, workingDirectory, null);
```

Comment:
I don't like that we have to use null here as well. Especially since `CacheContext` has a method `isCacheEnabled`, it feels wrong to me to have it nullable (as then we'd have to check for null, then check for this boolean to be able to use the cache context, which is too much in my opinion).

Maybe an implementation of a dummy cache could exist inside the frontend? Besides, having one here could mean we don't need one in the `python-checks-testkit` and you can use the one from the frontend instead to address my other 2 comments?